tests/test_benchmark_simple_scenarios.py

```diff
@@ -167,6 +167,56 @@ def test_benchmark_AD_grid_diesel(self, margs):
             assert sum(busses_flow["Diesel generator"]) == sum(busses_flow["demand_01"])
 
     # todo: Add test for fuel consumption (kWh/l).
+
+    @pytest.mark.skipif(
+        EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
+        reason="Benchmark test deactivated, set env variable "
+        "EXECUTE_TESTS_ON to 'master' to run this test",
+    )
+    @mock.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace())
+    def test_benchmark_AE_grid_battery_peak_pricing(self, margs):
+        # define the two cases needed for comparison (grid + PV) and (grid + PV + battery)
+        use_case = "AE_grid_battery_peak_pricing"
+        # define an empty dictionary for excess electricity
+        main(
+            overwrite=True,
+            display_output="warning",
+            path_input_folder=os.path.join(TEST_INPUT_PATH, use_case),
+            input_type=CSV_EXT,
+            path_output_folder=os.path.join(TEST_OUTPUT_PATH, use_case),
+        )
+        # read json with results file
+        with open(
+            os.path.join(TEST_OUTPUT_PATH, use_case, "json_with_results.json"), "r"
+        ) as results:
+            data = json.load(results)
+        peak_demand = data["energyConversion"][
+            "Electricity grid DSO_consumption_period_1"
+        ]["optimizedAddCap"]["value"]
+        # read timeseries_all_busses excel file
+        busses_flow = pd.read_excel(
+            os.path.join(TEST_OUTPUT_PATH, use_case, "timeseries_all_busses.xlsx"),
+            sheet_name=bus_suffix("Electricity"),
+        )
+        # make the time the index
+        busses_flow = busses_flow.set_index("Unnamed: 0")
+        # save the columns with the values to be used
+        DSO_period_1 = busses_flow["Electricity grid DSO_consumption_period_1"]
+        demand = busses_flow["demand_01"]
+        battery_charge = busses_flow["battery"]
+        battery_discharge = (
+            abs(busses_flow["demand_01"])
+            - busses_flow["Electricity grid DSO_consumption_period_1"]
+        )
+        # look for peak demand in column
+        for i in range(0, len(DSO_period_1)):
```

Comment:
I would have thought to only use few instances per DSO_period for asserting the validity - might be quicker. How long does the test take? 

Still, you also need to check DSO_period_2 and DSO_period_3 - what if I did a mistake defining them? DSO_period_1 would be 0 for the whole duration of period 2 and 3.

Reply:
okay I know for sure how to make it work for all the timestamps. At the moment I committed an attempt at choosing instances to test. I am really not so sure about the code and don't know how to implement it otherwise. If you're okay with evaluating all the timestamps, I can just replace one line of code for that.